### PR TITLE
feat: abort on missing file url

### PR DESCRIPTION
**Pull request by Chumper**
Allows the cache to return non-existing file urls

This required for tools like python that will compile the binaries when no pre-build binaries are found

### CODE REVIEW

**Comment by viceice** [comment]
We will remove the compile on the fly feature. It's no longer needed, but we should do a major bump for removal.

**Review by viceice — COMMENTED** [review]
As said, i don't think we need this. Makes it more complicated.

**Comment by Chumper** [comment]
Ok, so when the file cannot be donwloaded the cache path should be returned regardless?

**Comment by viceice** [comment]
Sure, if it exists. So it should fail if cache is missing and download fails.

**Comment by Chumper** [comment]
Ok, in that case we should merge this PR. It will return an empty path if the file does not exist in the cache or cannot be downloaded.

Or would you rather have it exit with a 1 status code?

**Comment by viceice** [comment]
Let's exit with error, so we don't need an extra check in tool code.

File: src/usr/local/buildpack/utils/cache.sh [hunk 1]
```diff
@@ -37,8 +40,9 @@ function download_file () {
   name=${2:-$(basename "${url}")}
 
   local temp_folder=${BUILDPACK_CACHE_DIR:-${TEMP_DIR}}
-  curl --create-dirs -sSfLo "${temp_folder}/${name}" "${url}"
-  echo "${temp_folder}/${name}"
+  if curl --create-dirs -sSfLo "${temp_folder}/${name}" "${url}" >/dev/null 2>&1 ; then
+    echo "${temp_folder}/${name}"
+  fi
```
Comment (viceice) [inline src/usr/local/buildpack/utils/cache.sh:45]: Didn't we aggreed download should fail all? 🤔

Comment (Chumper) [reply src/usr/local/buildpack/utils/cache.sh:45]: we did, but I haven't changed the line yet 🙈

**Pull request closed without merging**